File: DailyRPG_API_ASP.NET/Models/InventorySlot.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace DailyRpg.Models
{
    public class InventorySlot
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int HunterUserId { get; set; }

        [JsonIgnore]
        [ForeignKey("HunterUserId")]
        public virtual HunterUser HunterUser { get; set; } = null!;

        [Required]
        public int ItemId { get; set; }

        [ForeignKey("ItemId")]
        public virtual Item Item { get; set; } = null!;

        [Required]
        public int Quantity { get; set; }
    }
}

File: frontend/lib/models/inventory_slot.dart
import 'item.dart';


class InventorySlot {

  final int id;        
   int quantity; 

  final Item item;

  InventorySlot({
    required this.id,
    required this.quantity,
    required this.item,
  });

  factory InventorySlot.fromJson(Map<String, dynamic> json) {

    return InventorySlot(

      id: json['id'],
      quantity: json['quantity'],


      item: Item.fromJson(json['item']),
    );
  }
}